plemiona_api/management.py

```diff
@@ -60,3 +67,46 @@ def create_jwt(cls, user):
     def get_the_token_from_header(cls, token):
         token = token.replace("Bearer", "").replace(" ", "")  # clean the token
         return token
+
+@database_sync_to_async
+def get_player(validated_token):
+    try:
+        return Player.objects.get(id=validated_token["player_id"])
+
+    except Player.DoesNotExist:
+        return None
+
+
+class JwtAuthMiddleware(BaseMiddleware):
+    def __init__(self, inner):
+        self.inner = inner
+
+    async def __call__(self, scope, receive, send):
+        # Close old database connections to prevent usage of timed out connections
+        close_old_connections()
+
+        # Get the token
+        token = parse_qs(scope["query_string"].decode("utf8")).get("token")
+        if not token:
+            return await self.inner(scope, receive, send)
+
+        token = token[0]
+
+        # Try to authenticate the user
+        try:
+            # This will automatically validate the token and raise an error if token is invalid
+            UntypedToken(token)
+        except (InvalidToken, TokenError) as e:
+            # Token is invalid
+            return await self.inner(scope, receive, send)
+        else:
+            #  Then token is valid, decode it
+            decoded_data = jwt_decode(token, settings.SECRET_KEY, algorithms=["HS256"])
```

Comment:
You can use `jwt.deocde` here and remove importing this
```suggestion
            decoded_data = jwt.decode(token, settings.SECRET_KEY, algorithms=["HS256"])
```